ape_hardhat/providers.py

```diff
@@ -137,7 +137,7 @@ def process_name(self) -> str:
 
     @property
     def chain_id(self) -> int:
-        if hasattr(self._web3, "eth"):
```

Comment:
why is this being deleted?
I have seen cases where `eth` did not exist and that caused strange errors. I believe it is here for a reason!
If `mypy` is giving you issues, just use a `type: ignore` on it

Reply:
`self._web3` is `Optional[Web3]` so will need to handle the case where it is empty for mypy's sake. This was my way of doing it.

Reply:
> `self._web3` is `Optional[Web3]` so will need to handle the case where it is empty for mypy's sake. This was my way of doing it.

Yeah, but it just does not account for the situation where `_web3` is not None but `_web.eth` is None... I wrote about this on Telegram; not sure if the check is still needed but that was why it was put there in the first place.

Probably `type: ignore` is safer for now and then just switch to config-driven chain ID and include it in the CLI or config.

Reply:
You can also use `self.web3` instead of `self._web3`! 
However - I kind of like the idea of just always returning the hard-coded one for now.